ossl/src/digest.rs

```diff
@@ -37,6 +37,17 @@ impl EvpMd {
     pub unsafe fn as_mut_ptr(&mut self) -> *mut EVP_MD {
         self.ptr
     }
+
+    /// Tries to clone the `EvpMd`.
+    pub fn try_clone(&self) -> Result<EvpMd, Error> {
```

Comment:
Uhmm while from the rust structure side this is somewhat of a clone, it is actually a reference, so I would call it try_reference()

Reply:
If you look at reference-counted smart pointers in the Rust standard library, they also call this operation clone: https://doc.rust-lang.org/std/rc/struct.Rc.html#method.clone

Therefore, I think Rust programmers would expect the function to be called `try_clone`.

Reply:
Yes, ideally, but [`Clone::clone`](https://doc.rust-lang.org/std/clone/trait.Clone.html#tymethod.clone) is infallible.

There is precedence for `try_clone` in the standard library, e.g.: https://doc.rust-lang.org/std/fs/struct.File.html#method.try_clone

Reply:
we can make this call infallible too, the examples you provided also Panic!() in case of internal failure to clone()